frontend/src/components/common/private-route/private-route.tsx

```diff
@@ -22,6 +22,36 @@ const PrivateRoute: FC<Props> = ({
     return <Navigate to={redirectTo} />;
   }
 
+  const { pathname } = useLocation();
+
+  let requiredPermission: string;
+
+  switch (pathname) {
+    case AppRoute.EAM: {
+      requiredPermission = 'manage-eam';
+      break;
+    }
+    case AppRoute.BS: {
+      requiredPermission = 'manage-bs';
+      break;
+    }
+    case AppRoute.SC: {
+      requiredPermission = 'manage-sc';
+      break;
+    }
+    case AppRoute.SLC: {
+      requiredPermission = 'manage-slc';
+      break;
+    }
+    default: {
+      requiredPermission = '';
+    }
+  }
+
+  if (requiredPermission && !user?.permissions.includes(requiredPermission)) {
+    return <Navigate to={AppRoute.ROOT} />;
+  }
```

Comment:
```suggestion
```
you need to add new prop `permissions` (should be an array) to the `private-route` component and provide needed permissions in the `app` component for each route

in the `private-route` you need to compare permission the we pass through props and user's permissions